README.rst

```diff
@@ -466,4 +466,108 @@ Run tests with:
 
 .. code-block:: shell
 
+    # run qa checks
+    ./run-qa-checks
+
+    # standard tests
     ./runtests.py
+
+    # tests for the sample app
+    SAMPLE_APP=1 ./runtests.py --keepdb --failfast
+
+When running the last line of the previous example, the environment variable
+``SAMPLE_APP`` activates the sample apps in ``/tests/openwisp2/``
+which are simple django apps that extend ``openwisp-monitoring`` with
+the sole purpose of testing its extensibility, for more information regarding
+this concept, read the following section.
+
+Extending openwisp-monitoring
+-----------------------------
+
+The django apps in ``tests/openwisp2/`` namely ``sample_monitoring``, ``sample_check``,
+``sample_device_monitoring`` add some changes on top of the ``openwisp_monitoring`` module
+with the sole purpose of testing the module's extensibility.
+
+``sample_monitoring``, ``sample_check`` and ``sample_device_monitoring`` may serve as an example for
+extending *openwisp-monitoring* in your own application.
+
+All apps in *openwisp-monitoring* provide a set of models and admin classes which can
+be imported, extended and reused by third party apps.
+
+To extend any app in *openwisp-monitoring*, you **MUST NOT** add it to ``settings.INSTALLED_APPS``,
+but you must create your own app (which goes into ``settings.INSTALLED_APPS``), import the
+base classes from the respective app in ``openwisp-monitoring`` and add your customizations.
+
+In order to help django find the static files and templates of *openwisp-monitoring*,
+you need to perform the steps described below.
```

Comment:
Don't forget the premise mentioned here: https://github.com/openwisp/openwisp-firmware-upgrader#extending-openwisp-firmware-upgrader